appinventor/codelabs/i-have-a-dream/index.html

```diff
@@ -0,0 +1,235 @@
+        </ol>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Checkoff #2">
+        <aside class="special">
+          <h2>i-have-a-dream-2</h2>
+          <p><b>To recieve this checkoff, you must</b></p>
+          <ul class-"checklist">
+            <li>Show your completed user interface</li>
+            <li>Explain what the HorizontalArrangement component does</li>
+            <li>Explain the difference between an Image component and a Button component with an image</li>
+            <li>Give reasons why changing the name of the Player and Button components may be useful</li>
+          </ul>
+        </aside>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add code to pause MLK speech">
+        <p>Go to <b>Blocks View</b>. Let's add the ability to pause the speech if it is already playing. To do this, we will use an if-else block.</p>
+        <p>First, drag a <code>when MLKButton.Click</code> block. Then, drag an <code>if</code> block from the <b>Control</b> drawer. Click the blue modifier icon, and drag an <code>else</code> block into the picture of the <code>if</code> block. This modifies your <code>if</code> block to have two branches.</p>
+        <img src="img/if_else.png">
+        <p>Place the if-else block into the <code>when MLKButton.Click</code> event handler. Fill in the if-else block with the following blocks as shown.</p>
+        <img src="img/if_else_blocks.png">
+        <aside class="note">
+          <p>Conditional statements are fundamental to programming. They allow the program to make decisions based on the current state of the program. Almost all "if" statements check the values of component properties or values. In this case, the Player component's IsPlaying property is checked, and the program chooses an action based on whether the Player is playing or not.</p>
+        </aside>
+
+      </google-codelab-step>
+
+      <google-codelab-step label="Add code to pause the other speech">
+        <p>Now, we want to ensure that both speeches will never play at the same time. If the MLK speech is playing and we click the Malcolm X button, we want the MLK speech to pause.</p>
+        <p>To do this, all we need to do is add a <code>call MalcolmPlayer.Pause</code> block to the event handler as shown.</p>
+        <img src="img/pause_other.png">
+        <h2>Challenges</h2>
+        <ol>
+          <li>Code the Malcolm X button exactly like the MLK button. We want both buttons to behave the exact same way.</li>
+          <p><i>Hint: You can copy-paste blocks, as well as change the blocks to refer to different components using the arrow next to the component name.</i></p>
+        </ol>
+        <aside class="callout">
+          <h2>Test your app</h2>
+          <p>Try playing the two speeches. Can you pause a speech once it's playing? If a speech is playing, will clicking the other button pause that speech and start the other speech?</p>
```

Comment:
Try playing and pausing the two speeches. Start MLK's speech. Before it ends, click on Malcolm X. What happens?